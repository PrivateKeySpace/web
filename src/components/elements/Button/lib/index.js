export { default as mapProps } from './mapProps'
export { default as propTypes } from './propTypes'
