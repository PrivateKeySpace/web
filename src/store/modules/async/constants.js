export const STATUS_PENDING = 'pending'
export const STATUS_SUCCESS = 'success'
export const STATUS_FAILURE = 'failure'

export const METHOD_GET = 'GET'
export const METHOD_POST = 'POST'
export const METHOD_PATCH = 'PATCH'
export const METHOD_PUT = 'PUT'
