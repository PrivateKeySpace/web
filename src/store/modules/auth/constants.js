export const STORAGE_RECORD_KEY = 'auth'

// One hour TTL, subject to change
export const STORAGE_RECORD_TTL = 60 * 60 * 1000
