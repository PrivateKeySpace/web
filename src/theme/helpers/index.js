
export { default as aphroditeGlobal, GLOBAL_STYLES_SELECTOR } from './aphroditeGlobal'
export { default as getMediaQueries } from './getMediaQueries'
export { default as mediaQuerizeUtil } from './mediaQuerizeUtil'
