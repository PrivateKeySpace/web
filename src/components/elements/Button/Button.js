import React from 'react'
import { compose, setDisplayName, setPropTypes, mapProps, pure } from 'recompose'
import { propTypes, mapProps as propsMapper } from './lib'

const displayName = 'components/elements/Button'
const enhance = compose(
  setDisplayName(displayName),
  setPropTypes(propTypes),
  mapProps(propsMapper),
  pure
)

function Button (props) {
  return <button {...props} />
}

export default enhance(Button)
