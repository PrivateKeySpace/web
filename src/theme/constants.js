import { getMediaQueries } from './helpers'

export const breakpoints = {
  xs: 500,
  sm: 768,
  md: 1024,
  lg: 1280,
  xl: 1400
}

export const mediaQueries = getMediaQueries(breakpoints)

export const scale = [0, 5, 10, 15, 25]

export const colors = {
  black: '#000000',
  white: '#ffffff',
  gray: '#666666',
  darkBg: '#323a3e',
  lightBg: '#f3f3f3',
  danger: '#f45c3e',
  warning: '#fcca60',
  success: '#76ab26',
  info: '#3ba6d8'
}

export const fontSizes = []
export const lineHeights = []

export const letterSpacings = []

export const fontFamilies = { primary: 'Helvetica, Segoe UI, sans-serif' }

export const borderColor = colors.black
export const borderWidth = 1
export const borderStyle = 'solid'
export const borderRadius = 5
