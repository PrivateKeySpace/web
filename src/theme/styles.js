import './stylesGlobal'
import { StyleSheet } from 'aphrodite/no-important'
import { mediaQuerizeUtil } from './helpers'
import { breakpoints, mediaQueries, borderStyle, borderColor, borderWidth, borderRadius, scale, colors,
  fontFamilies, fontSizes, lineHeights } from './constants'

let stylesSpec = {}

stylesSpec.inlineFlex = { display: 'inline-flex' }
stylesSpec.flex = { display: 'flex' }
stylesSpec.flexWrap = { flexWrap: 'wrap' }
stylesSpec.flexDirectionColumn = { flexDirection: 'column' }
stylesSpec.flexDirectionRow = { flexDirection: 'row' }
stylesSpec.flexGrow1 = { flexGrow: 1 }
stylesSpec.flexShrink1 = { flexShrink: 1 }
stylesSpec.flexGrow0 = { flexGrow: 0 }
stylesSpec.flexShrink0 = { flexShrink: 0 }
stylesSpec.alignItemsCenter = { alignItems: 'center' }
stylesSpec.alignItemsFlexEnd = { alignItems: 'flex-end' }
stylesSpec.alignItemsFlexStart = { alignItems: 'flex-start' }
stylesSpec.justifyContentCenter = { justifyContent: 'center' }
stylesSpec.justifyContentSpaceAround = { justifyContent: 'space-around' }
stylesSpec.justifyContentSpaceBetween = { justifyContent: 'space-between' }
stylesSpec.justifyContentFlexStart = { justifyContent: 'flex-start' }
stylesSpec.justifyContentFlexEnd = { justifyContent: 'flex-end' }

stylesSpec.width100 = { width: '100%' }
stylesSpec.height100 = { height: '100%' }

/*
.fontFamily{0-1}
 */
{
  const fontFamiliesNames = Object.keys(fontFamilies)
  fontFamiliesNames.forEach(fontFamilyName => {
    stylesSpec[`fontFamily${fontFamilyName}`] = { fontFamily: fontFamilies[fontFamilyName] }
  })
}

/*
.fontWeight{100-900}
 */
{
  const fontWeights = [100, 200, 300, 400, 500, 600, 700, 800, 900]
  fontWeights.forEach(fontWeight => {
    stylesSpec[`fontWeight${fontWeight}`] = { fontWeight }
  })
}

/*
.fontSize{0-7}
.fontSize{0-7}_{xs|sm|md|lg|xl}
 */
fontSizes.forEach((fontSize, index) => {
  const props = {
    fontSize: `${fontSize}px`,
    lineHeight: `${lineHeights[index]}px`
  }

  stylesSpec[`fontSize${index}`] = props
  stylesSpec = mediaQuerizeUtil(stylesSpec, props, breakpointName => `fontSize${index}_${breakpointName}`)
})

stylesSpec.textAlignCenter = { textAlign: 'center' }
stylesSpec.textAlignLeft = { textAlign: 'left' }
stylesSpec.textAlignRight = { textAlign: 'right' }
stylesSpec.textTransformUppercase = { textTransform: 'uppercase' }

stylesSpec.displayBlock = { display: 'block' }
stylesSpec.displayInlineBlock = { display: 'inline-block' }

stylesSpec.overflowHidden = { overflow: 'hidden' }

stylesSpec.hidden = { display: 'none' }

/*
.{hidden|visible}_{xs|sm|md|lg|xl}
 */
{
  const breakpointsNames = Object.keys(breakpoints)
  breakpointsNames.forEach(breakpointName => {
    stylesSpec[`hidden_${breakpointName}`] = { [mediaQueries[`max_${breakpointName}`]]: { display: 'none' } }
    stylesSpec[`visible_${breakpointName}`] = { [mediaQueries[`max_${breakpointName}`]]: { display: 'inherit' } }
  })
}

stylesSpec.hidden_print = { [mediaQueries.print]: { display: 'none' } }

stylesSpec.cursorPointer = { cursor: 'pointer' }

stylesSpec.cursorMove = { cursor: 'move' }

/*
.col{1-12}
.col{1-12}_{xs|sm|md|lg|xl}
 */
{
  const cols = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
  cols.forEach(col => {
    const props = { width: `${(col / cols.length) * 100}%` }

    stylesSpec[`col${col}`] = props
    stylesSpec = mediaQuerizeUtil(stylesSpec, props, breakpointName => `col${col}_${breakpointName}`)
  })
}

stylesSpec.border = { borderWidth: `${borderWidth}px`, borderStyle, borderColor }
stylesSpec.noBorder = { borderStyle: 'none' }
stylesSpec.borderRadius = { borderRadius }

/*
.border{Top|Right|Bottom|Left}Width
.border{Top|Right|Bottom|Left}Style
.border{Top|Right|Bottom|Left}Color
 */
{
  const directions = ['Top', 'Right', 'Bottom', 'Left']
  directions.forEach(direction => {
    stylesSpec[`border${direction}`] = {
      [`border${direction}Width`]: `${borderWidth}px`,
      [`border${direction}Style`]: borderStyle,
      [`border${direction}Color`]: borderColor
    }
  })
}

/*
.borderColor_{color}
.backgroundColor_{color}
.color_{color}
 */
{
  const propColorList = ['borderColor', 'backgroundColor', 'color']
  Object.keys(colors).forEach(color => {
    propColorList.forEach(prop => {
      stylesSpec[`${prop}_${color}`] = { [prop]: colors[color] }
    })
  })
}

/*
.{m|p}{t|r|b|l|x|y}{0-4}_{xs|sm|md|lg|xl}
 */
{
  const marginPropsKeys = ['m', 'mt', 'mr', 'mb', 'ml', 'mx', 'my']
  const paddingPropsKeys = ['p', 'pt', 'pr', 'pb', 'pl', 'px', 'py']

  function keyToScaleProp (propName, scaleIndex, scale) {
    return (
      (typeof scaleIndex === 'number')
        ? { [propName]: `${scale[scaleIndex]}px` }
        : null
    )
  }

  function keysToMarginScaleProps (keys, scale) {
    const { m, mt, mb, my, ml, mr, mx } = keys
    return {
      ...keyToScaleProp('margin', m, scale),
      ...keyToScaleProp('marginTop', mt, scale),
      ...keyToScaleProp('marginBottom', mb, scale),
      ...keyToScaleProp('marginTop', my, scale),
      ...keyToScaleProp('marginBottom', my, scale),
      ...keyToScaleProp('marginLeft', ml, scale),
      ...keyToScaleProp('marginRight', mr, scale),
      ...keyToScaleProp('marginLeft', mx, scale),
      ...keyToScaleProp('marginRight', mx, scale)
    }
  }

  function keysToPaddingScaleProps (keys, scale) {
    const { p, pt, pb, py, pl, pr, px } = keys
    return {
      ...keyToScaleProp('padding', p, scale),
      ...keyToScaleProp('paddingTop', pt, scale),
      ...keyToScaleProp('paddingBottom', pb, scale),
      ...keyToScaleProp('paddingTop', py, scale),
      ...keyToScaleProp('paddingBottom', py, scale),
      ...keyToScaleProp('paddingLeft', pl, scale),
      ...keyToScaleProp('paddingRight', pr, scale),
      ...keyToScaleProp('paddingLeft', px, scale),
      ...keyToScaleProp('paddingRight', px, scale)
    }
  }

  scale.forEach((value, index) => {
    marginPropsKeys.forEach(key => {
      const props = keysToMarginScaleProps({ [key]: index }, scale)
      stylesSpec[`${key}${index}`] = props
      stylesSpec = mediaQuerizeUtil(stylesSpec, props, breakpointName => `${key}${index}_${breakpointName}`)
    })
    paddingPropsKeys.forEach(key => {
      const props = keysToPaddingScaleProps({ [key]: index }, scale)
      stylesSpec[`${key}${index}`] = props
      stylesSpec = mediaQuerizeUtil(stylesSpec, props, breakpointName => `${key}${index}_${breakpointName}`)
    })
  })
}

const styles = StyleSheet.create(stylesSpec)

export default styles
