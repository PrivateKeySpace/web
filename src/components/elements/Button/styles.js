import { StyleSheet } from 'aphrodite/no-important'
import { darken } from 'polished'
import { colors, variants } from '../../../theme/constants'

const buttonHeight = 38

const buttonStyle = {
  height: buttonHeight,
  minWidth: 38,
  borderRadius: buttonHeight,
  lineHeight: `${buttonHeight}px`,
  fontSize: 13,
  cursor: 'pointer',
  userSelect: 'none',
  outline: 'none',
  paddingLeft: 16,
  paddingRight: 16,
  ':disabled': {
    opacity: 0.5,
    cursor: 'not-allowed'
  }
}

const roundButtonStyle = {
  width: 38,
  paddingLeft: 1,
  paddingRight: 0
}

export const defaultSx = StyleSheet.create({
  button: {
    ...buttonStyle,
    backgroundColor: colors.lightGray,
    border: 'none',
    color: colors.black
  },
  round: roundButtonStyle,
  ...variants.reduce((acc, variant) => {
    acc[variant] = {
      color: colors.white,
      backgroundColor: colors[variant],
      ':hover:enabled': {
        backgroundColor: darken(0.08, colors[variant])
      },
      ':active:enabled': {
        backgroundColor: darken(0.12, colors[variant])
      }
    }
    return acc
  }, {})
})

export const outlineSx = StyleSheet.create({
  button: {
    ...buttonStyle,
    background: 'transparent',
    borderWidth: 2,
    borderStyle: 'solid',
    borderColor: colors.black,
    color: colors.black,
    paddingLeft: 14,
    paddingRight: 14
  },
  round: roundButtonStyle,
  ...variants.reduce((acc, variant) => {
    acc[variant] = {
      borderColor: colors[variant],
      color: colors[variant],
      ':hover:enabled': {
        borderColor: darken(0.08, colors[variant]),
        backgroundColor: darken(0.08, colors[variant]),
        color: colors.white
      },
      ':active:enabled': {
        borderColor: darken(0.12, colors[variant]),
        backgroundColor: darken(0.12, colors[variant]),
        color: colors.white
      }
    }
    return acc
  }, {})
})

export const linkSx = StyleSheet.create({
  button: {
    ...buttonStyle,
    background: 'transparent',
    border: 'none',
    color: colors.black
  },
  round: roundButtonStyle,
  ...variants.reduce((acc, variant) => {
    acc[variant] = {
      color: colors[variant],
      ':hover:enabled': {
        color: darken(0.08, colors[variant])
      },
      ':active:enabled': {
        color: darken(0.12, colors[variant])
      }
    }
    return acc
  }, {})
})
