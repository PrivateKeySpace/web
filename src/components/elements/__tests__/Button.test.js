import React from 'react'
import { StyleSheetTestUtils } from 'aphrodite'
import { render } from 'enzyme'
import { variants } from '../../../theme/constants'
import Button from '../Button'

beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection()
})

afterEach(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
})

it('renders default style buttons of all variants without crashing and matches snapshot', () => {
  for (const variant of variants) {
    const button = render(
      <Button variant={variant}>Test Button</Button>
    )
    expect(button).toMatchSnapshot()
  }
})

it('renders disabled default style buttons of all variants without crashing and matches snapshot', () => {
  for (const variant of variants) {
    const button = render(
      <Button variant={variant} disabled>Test Button</Button>
    )
    expect(button).toMatchSnapshot()
  }
})

it('renders outline style buttons of all variants without crashing and matches snapshot', () => {
  for (const variant of variants) {
    const button = render(
      <Button variant={variant} outline>Test Button</Button>
    )
    expect(button).toMatchSnapshot()
  }
})

it('renders link style buttons of all variants without crashing and matches snapshot', () => {
  for (const variant of variants) {
    const button = render(
      <Button variant={variant} link>Test Button</Button>
    )
    expect(button).toMatchSnapshot()
  }
})

it('renders round style buttons of all variants without crashing and matches snapshot', () => {
  for (const variant of variants) {
    const button = render(
      <Button variant={variant} round>
        <span className='lnr lnr-alarm' />
      </Button>
    )
    expect(button).toMatchSnapshot()
  }
})

it('renders round ouline style buttons of all variants without crashing and matches snapshot', () => {
  for (const variant of variants) {
    const button = render(
      <Button variant={variant} round outline>
        <span className='lnr lnr-alarm' />
      </Button>
    )
    expect(button).toMatchSnapshot()
  }
})
