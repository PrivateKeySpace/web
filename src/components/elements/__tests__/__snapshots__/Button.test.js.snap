// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default style buttons of all variants without crashing and matches snapshot 1`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_v3fs7o-o_O-danger_ut56jk"
>
  Test Button
</button>
`;

exports[`renders default style buttons of all variants without crashing and matches snapshot 2`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_v3fs7o-o_O-success_ho83a3"
>
  Test Button
</button>
`;

exports[`renders default style buttons of all variants without crashing and matches snapshot 3`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_v3fs7o-o_O-warning_1d2e13p"
>
  Test Button
</button>
`;

exports[`renders default style buttons of all variants without crashing and matches snapshot 4`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_v3fs7o-o_O-info_4c6fs9"
>
  Test Button
</button>
`;

exports[`renders disabled default style buttons of all variants without crashing and matches snapshot 1`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_v3fs7o-o_O-danger_ut56jk"
  disabled=""
>
  Test Button
</button>
`;

exports[`renders disabled default style buttons of all variants without crashing and matches snapshot 2`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_v3fs7o-o_O-success_ho83a3"
  disabled=""
>
  Test Button
</button>
`;

exports[`renders disabled default style buttons of all variants without crashing and matches snapshot 3`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_v3fs7o-o_O-warning_1d2e13p"
  disabled=""
>
  Test Button
</button>
`;

exports[`renders disabled default style buttons of all variants without crashing and matches snapshot 4`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_v3fs7o-o_O-info_4c6fs9"
  disabled=""
>
  Test Button
</button>
`;

exports[`renders link style buttons of all variants without crashing and matches snapshot 1`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_olhiji-o_O-danger_1apl0py"
>
  Test Button
</button>
`;

exports[`renders link style buttons of all variants without crashing and matches snapshot 2`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_olhiji-o_O-success_1sf415p"
>
  Test Button
</button>
`;

exports[`renders link style buttons of all variants without crashing and matches snapshot 3`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_olhiji-o_O-warning_1nw1vbn"
>
  Test Button
</button>
`;

exports[`renders link style buttons of all variants without crashing and matches snapshot 4`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_olhiji-o_O-info_19dksof"
>
  Test Button
</button>
`;

exports[`renders outline style buttons of all variants without crashing and matches snapshot 1`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1yqkn2s-o_O-danger_6ixs7d"
>
  Test Button
</button>
`;

exports[`renders outline style buttons of all variants without crashing and matches snapshot 2`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1yqkn2s-o_O-success_rpvc09"
>
  Test Button
</button>
`;

exports[`renders outline style buttons of all variants without crashing and matches snapshot 3`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1yqkn2s-o_O-warning_1ijooeh"
>
  Test Button
</button>
`;

exports[`renders outline style buttons of all variants without crashing and matches snapshot 4`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1yqkn2s-o_O-info_1fyhmo9"
>
  Test Button
</button>
`;

exports[`renders round ouline style buttons of all variants without crashing and matches snapshot 1`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1yqkn2s-o_O-danger_6ixs7d-o_O-fontSize1_3kvou9-o_O-fontWeight500_gt7myn-o_O-round_g0qzym"
>
  <span
    class="lnr lnr-alarm"
  >
    
  </span>
</button>
`;

exports[`renders round ouline style buttons of all variants without crashing and matches snapshot 2`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1yqkn2s-o_O-success_rpvc09-o_O-fontSize1_3kvou9-o_O-fontWeight500_gt7myn-o_O-round_g0qzym"
>
  <span
    class="lnr lnr-alarm"
  >
    
  </span>
</button>
`;

exports[`renders round ouline style buttons of all variants without crashing and matches snapshot 3`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1yqkn2s-o_O-warning_1ijooeh-o_O-fontSize1_3kvou9-o_O-fontWeight500_gt7myn-o_O-round_g0qzym"
>
  <span
    class="lnr lnr-alarm"
  >
    
  </span>
</button>
`;

exports[`renders round ouline style buttons of all variants without crashing and matches snapshot 4`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1yqkn2s-o_O-info_1fyhmo9-o_O-fontSize1_3kvou9-o_O-fontWeight500_gt7myn-o_O-round_g0qzym"
>
  <span
    class="lnr lnr-alarm"
  >
    
  </span>
</button>
`;

exports[`renders round style buttons of all variants without crashing and matches snapshot 1`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_v3fs7o-o_O-danger_ut56jk-o_O-fontSize1_3kvou9-o_O-fontWeight500_gt7myn-o_O-round_g0qzym"
>
  <span
    class="lnr lnr-alarm"
  >
    
  </span>
</button>
`;

exports[`renders round style buttons of all variants without crashing and matches snapshot 2`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_v3fs7o-o_O-success_ho83a3-o_O-fontSize1_3kvou9-o_O-fontWeight500_gt7myn-o_O-round_g0qzym"
>
  <span
    class="lnr lnr-alarm"
  >
    
  </span>
</button>
`;

exports[`renders round style buttons of all variants without crashing and matches snapshot 3`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_v3fs7o-o_O-warning_1d2e13p-o_O-fontSize1_3kvou9-o_O-fontWeight500_gt7myn-o_O-round_g0qzym"
>
  <span
    class="lnr lnr-alarm"
  >
    
  </span>
</button>
`;

exports[`renders round style buttons of all variants without crashing and matches snapshot 4`] = `
<button
  class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_v3fs7o-o_O-info_4c6fs9-o_O-fontSize1_3kvou9-o_O-fontWeight500_gt7myn-o_O-round_g0qzym"
>
  <span
    class="lnr lnr-alarm"
  >
    
  </span>
</button>
`;
